query-engine/request-handlers/src/handler.rs

```diff
@@ -172,7 +174,15 @@ impl<'a> RequestHandler<'a> {
                                 responses.insert_data(&singular_name, Item::Map(result));
                             }
                             _ => {
-                                responses.insert_data(&singular_name, Item::null());
+                                if reject_on_not_found {
+                                    responses.insert_error(GQLError::from_user_facing_error(
+                                        user_facing_errors::Error::from(KnownError::new(RecordRequiredButNotFound {
+                                            cause: "Expected a record, found none.".to_owned(),
+                                        })),
+                                    ));
```

Comment:
this error seems to be missing a couple layers of wrapping, but I wasn't able to figure out the how to correctly reproduce this error